hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java

```diff
@@ -92,7 +93,7 @@ private void doSendResponse(Object value) {
                 sendResponse(value);
             } catch (Exception e) {
                 Throwable ex = peel(e);
-                if (value instanceof Throwable && ex instanceof HazelcastSerializationException) {
+                if (value instanceof Throwable && isOrHasCause(ex, HazelcastSerializationException.class)) {
```

Comment:
Could you please explain how we get a nested `HazelcastSerializationException`. Note that this exception is from `sendResponse`.